### PR TITLE
Introduce Session Timeout

### PR DESCRIPTION
**Pull request by MrGeneration**
This PR introduces the Session Timeout functionality to the documentation and also reworks the general security parts.

### CODE REVIEW

File: settings/security.rst [hunk 1]
```diff
@@ -1,90 +1,11 @@
...
-
-.. hint:: Normally, especially when using LDAP, you might want to consider to disable both above options, because you'd normally do that via LDAP.
-
-
-Password
---------
-
-This section allows you to define password requirements for the local user accounts.
-
-.. note:: Zammad does **not** allow you to change your LDAP password, instead, it will set a password in it's local database which might confuse your users.
```
Comment (techerella) [inline settings/security.rst:21]: "in its local database" (kein Apostroph)

File: settings/security.rst [hunk 2]
```diff
@@ -1,90 +1,11 @@
...
-^^^^^^^^^^^^^^^^^^^^^
-
-You can choose a value between *4* and *20*. This defines how often a login to a user account may fail until Zammad will lock it.
-Please note that via UI the only way to unlock a user account is to change the password (either as admin or via password reset function (if enabled)).
-The default value is `10`.
-
-.. note:: You can also unlock an account via `console <https://docs.zammad.org/en/latest/console/working-on-users.html>`_ or `API <https://docs.zammad.org/en/latest/api-user.html>`_.
-
-
-2 lower and 2 upper characters
```
Comment (techerella) [inline settings/security.rst:35]: 2 lower and 2 upper **case** characters

File: settings/security.rst [hunk 3]
```diff
@@ -1,90 +1,11 @@
...
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-You can add complexity into passwords by enforcing at least 2 upper and lower case characters. The default value is `no`.
```
Comment (techerella) [inline settings/security.rst:38]: "add complexity **to** passwords"

File: settings/security.rst [hunk 4]
```diff
@@ -1,90 +1,11 @@
...
-Minimum length
-^^^^^^^^^^^^^^
-
-This defines the minimum password length required (from *4* to *20*). The default value is `6`.
-
-
-Digit required
-^^^^^^^^^^^^^^
-
-This enforces your users to at least use one digit with his password.
```
Comment (techerella) [inline settings/security.rst:50]: with **their** password

File: settings/security.rst [hunk 5]
```diff
@@ -1,90 +1,11 @@
...
-The default value is `yes`.
-
-.. _third-party-logins:
-
-Third-Party Applications
-------------------------
-
-Third party authentication is a great way to help your users to login to Zammad easier.
-If the account is yet unknown, Zammad will create a new user automatically, without the user needed to interact (e.g. type in his name).
-Another big advantage of this feature is that your user doesn't need another password to remember.
```
Comment (techerella) [inline settings/security.rst:60]: doesn't need to remember another password

File: settings/security.rst [hunk 6]
```diff
@@ -1,90 +1,11 @@
...
-   security/third-party/saml
-
-.. note:: We're currently missing documentation for the following login providers:
-
-   * LinkedIn
-   * Weibo
-
-.. _automatic-account-linking:
-
-Automatic account link on initial logon
```
Comment (techerella) [inline settings/security.rst:81]: initial log**i**n?

File: settings/security.rst [hunk 7]
```diff
@@ -1,90 +1,11 @@
...
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-In general there's two ways two possible ways to handle how Zammad deals with users it already knows (from whatever source) when you try to authenticate
```
Comment (techerella) [inline settings/security.rst:84]: two ways two possible ways  --> two possible ways

File: settings/security.rst [hunk 8]
```diff
@@ -1,90 +1,11 @@
...
-against a third-party application. By default, Zammad will not automatically link "unknown" authentication providers to existing accounts.
-This means that the user has to manually link authentication providers to their accounts (for more about this `consult the user documentation <https://user-docs.zammad.org/en/latest/extras/profile-and-settings.html>`_).
-
-Sometimes, this doesn't come in handy, as this also means if you're trying to authenticate against a (yet) unknown third party authentication, Zammad will throw an error telling you, that the email address is already in use. If you want to ensure that no matter if your user links his account or not, you might want to enable "Automatic account link on initial logon"
```
Comment (techerella) [inline settings/security.rst:88]: ...telling you that the email address is already in use (kein Komma)

File: settings/security/base.rst [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+   User accounts are a direct dependency of tickets and thus technically
+   mandatory.
+
+Lost Password
+-------------
+
+.. container:: cfloat-left
+
+   Activates the `lost password` function on the login page.
+   If set to ``no`` only administrators may change the users password - users
```
Comment (techerella) [inline settings/security/base.rst:37]: ..." change the user's password"

File: settings/security/base.rst [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+   With third party authentications – but especially LDAP – you may want to
+   disable this function. Zammad **will not** change third party login
+   passwords and instead set or change the local password!
+
+Session Timeout
+---------------
+
+All settings below by default are set to ``4 weeks``.
+Session Timeout defines the life time of a users session.
+As soon reached, Zammad will automatically log off the session in question.
```
Comment (techerella) [inline settings/security/base.rst:61]: As soon as it is reached

File: settings/security/base.rst [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+
+Zammad takes the highest value set assigned for the user based on
+the permissions.
+
+   #. ``admin``
+   #. ``ticket.agent``
+   #. ``ticket.customer``
+   #. ``default`` (fallback if user doesn't have above permissions set)
+
+All settings act independent on each other allowing you to disable the
```
Comment (techerella) [inline settings/security/base.rst:71]: All settings act independently from each other

File: settings/security/password.rst [hunk 12]
```diff
@@ -0,0 +1,71 @@
+Password
+********
+
+This section allows you to define password requirements for the local user
+accounts.
+
+.. note:: 
+
+   Zammad does **not** allow you to change your LDAP password, instead, it will
+   set a password in it's local database which might confuse your users. This
```
Comment (techerella) [inline settings/security/password.rst:10]: ... "in its local database"

File: settings/security/password.rst [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+   will be addressed in the future by
+   `#1169 <https://github.com/zammad/zammad/issues/1169>`_ and
+   `#2389 <https://github.com/zammad/zammad/issues/2389>`_.
+
+.. warning:: **💪 Exception for strong passwords 💪**
+   
+   Please note that below password policies *do not* affect administrators
+   setting passwords on user accounts. While this seems strange and not safe
+   we believe that an administrator knowing an users passwords is insecure
```
Comment (techerella) [inline settings/security/password.rst:19]: ..."knowing a user's password"

File: settings/security/password.rst [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+   as well.
+
+   The suggested workflow is either:
+
+      * to use third party logins to not require local passwords at all
+        **- or -**
+      * to require your *user to reset the password* upon first login. 
+   
+   This way administrators are not required to set a users password at all!
```
Comment (techerella) [inline settings/security/password.rst:28]: "to set a user's password at all" (mit Apostroph)

File: settings/security/password.rst [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+The default value is ``10``.
+
+.. note:: 
+
+   You can also unlock an account via 
+   `console <https://docs.zammad.org/en/latest/console/working-on-users.html>`_
+   or `API <https://docs.zammad.org/en/latest/api-user.html>`_.
+
+
+2 lower and 2 upper characters
```
Comment (techerella) [inline settings/security/password.rst:48]: 2 lower and 2 upper **case** characters

File: settings/security/password.rst [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+------------------------------
+
+You can add complexity into passwords by enforcing at least 2 upper and lower
```
Comment (techerella) [inline settings/security/password.rst:51]: "add complexity **to** passwords"

File: settings/security/third-party.rst [hunk 17]
```diff
@@ -0,0 +1,50 @@
+Third-Party Applications
+************************
+
+Third party authentication is a great way to help your users to login to Zammad
```
Comment (techerella) [inline settings/security/third-party.rst:4]: Third-party authentication is a great way to help your users to log in to Zammad
more easily

File: settings/security/third-party.rst [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+easier. If the account is yet unknown, Zammad will create a new user
+automatically, without the user needed to interact (e.g. type in his name).
+Another big advantage of this feature is that your user doesn't need another
+password to remember.
```
Comment (techerella) [inline settings/security/third-party.rst:8]: doesn't need to remember another password

File: settings/security/third-party.rst [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+.. note:: 
+
+   We're currently missing documentation for the following login providers:
+
+      * LinkedIn
+      * Weibo
+
+.. _automatic-account-linking:
+
+Automatic account link on initial logon
```
Comment (techerella) [inline settings/security/third-party.rst:31]: logon oder login?

File: settings/security/third-party.rst [hunk 20]
```diff
@@ -0,0 +1,50 @@
...
+
+This means that the user has to manually link authentication providers to their
+accounts (for more about this 
+`consult the user documentation <https://user-docs.zammad.org/en/latest/extras/profile-and-settings.html>`_).
+
+Sometimes this doesn't come in handy as this also means you'll receive
+error messages about "email address being in use already" for (yet) unknown
+third-party authentication methods.
+
+If you want to allow your users to always being able to login, no matter what,
```
Comment (techerella) [inline settings/security/third-party.rst:47]: "always be able to log in"

**Review by techerella — APPROVED** [review]

File: settings/security/password.rst (hunk 15 above)
Comment (MrGeneration) [reply settings/security/password.rst:48]: Sorry, in this case this is not a typo but the settings actual name. 
I can't adjust the heading thus.

image

File: settings/security/third-party.rst (hunk 19 above)
Comment (MrGeneration) [reply settings/security/third-party.rst:31]: The setting contains logon:
image

**Pull request merged**